core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/JWTAuthenticationFilterTest.java

```diff
@@ -0,0 +1,82 @@
+@ActiveProfiles({"test", "security"})
+
+public class JWTAuthenticationFilterTest {
+
+    @Autowired
+    private JWTAuthenticationConverter jwtAuthenticationConverter;
+
+    @Test
+    public void testJWTDecodeException() throws Exception {
+
+        String token = "eyJ0eXAiOJKV1QiLCUzI1NiJ9.eyJzdWIiOiJMyIiwiaXNzIjoiaWVzaSIsImV4cCI6MTU5ODU2ODM2MywiaWF0IjoxNTk4NDY4MzYzLCJhdXRob3JpdGllcyI6W119.alYmNwRyiqS8VijXdQ2F0ooVdr5KkITz2bycffG6_fI";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.JWTDecodeException: The string");
+    }
+
+    @Test
+    public void testInvalidClaims() throws Exception {
+
+        String token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE0Nzc1OTJ9.isvT0Pqx0yjnZk53mUFSeYFJLDs-Ls9IsNAm86gIdZo";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.InvalidClaimException: The Claim");
+    }
+
+    @Test
+    public void testSignatureVerificationException() throws Exception {
+
+        String token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXUyJ9.eyJpc3MiOiJhdXRoMCJ9.AbIJTDMFc75MhvcP03nJPyCPzZtQcGEp-zWfOkEE";
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/scripts");
+        request.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
+        JWTAuthenticationFilter jwtAuthenticationFilter = new JWTAuthenticationFilter(jwtAuthenticationConverter);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        MockFilterChain chain = new MockFilterChain();
+
+        jwtAuthenticationFilter.doFilterInternal(request, response, chain);
+        assertThat(response.getStatus()).isEqualTo(401);
+        assertThat(response.getContentAsString()).contains("com.auth0.jwt.exceptions.SignatureVerificationException: The Token's Signature resulted invalid when verified using the Algorithm: HmacSHA256");
```

Comment:
is it possible to do a jsonPath assertion on the json string? As can you check that the 'message' field starts with "com.auth0..." and the 'status' field is equal to 401.